### PR TITLE
[typescript-language-features] Add suggest.allowIncompleteCompletions and pass trigger kind to TS Server

### PR DESCRIPTION
**Pull request by andrewbranch**
Counterpart to https://github.com/microsoft/TypeScript/pull/44713

### CODE REVIEW

**Comment by andrewbranch** [comment]
To discuss: how/whether to present `suggest.allowIncompleteCompletions` in settings UI

File: extensions/typescript-language-features/src/utils/typeConverters.ts [hunk 1]
```diff
@@ -126,3 +126,10 @@ export namespace SymbolKind {
 		}
 	}
 }
+
+export namespace CompletionTriggerKind {
+	// @ts-expect-error
+	export function toProtocolCompletionTriggerKind(kind: vscode.CompletionTriggerKind): Proto.CompletionTriggerKind {
+		return kind + 1;
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/utils/typeConverters.ts:133]: Can we use a more explicit mapping here, such as a switch statement or map?

File: extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts [hunk 2]
```diff
@@ -185,6 +185,8 @@ export default class FileConfigurationManager extends Disposable {
 			generateReturnInDocTemplate: config.get<boolean>('suggest.jsdoc.generateReturns', true),
 			includeCompletionsForImportStatements: config.get<boolean>('suggest.includeCompletionsForImportStatements', true),
 			includeCompletionsWithSnippetText: config.get<boolean>('suggest.includeCompletionsWithSnippetText', true),
+			// @ts-expect-error until 4.4
+			allowIncompleteCompletions: config.get<boolean>('suggest.allowIncompleteCompletions', true),
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts:189]: Following up on our discussion earlier today:  we agreed that this should not be user configurable

**Review by mjbvz — CHANGES_REQUESTED** [review]

**Comment by mjbvz** [comment]
Thanks!

**Pull request merged**